pkg/controller/core/workload_controller.go

```diff
@@ -159,6 +159,15 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
+	if features.Enabled(features.TopologyAwareScheduling) {
+		if !r.cache.TASCache().SyncedFlavors() {
+			log.V(2).Info("Waiting for TAS cache to be synced")
+			return ctrl.Result{
+				RequeueAfter: 1 * time.Second,
```

Comment:
I think it is fair to just return error, it will get retried by controller-runtime